### PR TITLE
Choose the right centos sku in azure

### LINKED ISSUE #1040: Azure CentOS 7 image is not available anymore
The CentOS 7 image that we use on Azure is not available anymore. Creating a CentOS machine causes the following error:

```
  Warning  ReconcilingError  37s (x25 over 18m)  machine_controller  failed to create machine at cloudprovider, due to trying to create a VM: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=404 -- Original Error: Code="PlatformImageNotFound" Message="The platform image 'OpenLogic:CentOS:7-CI:latest' is not available. Verify that all fields in the storage profile are correct. For more details about storage profile information, please refer to https://aka.ms/storageprofile" Target="imageReference"
```

This also causes the E2E tests to fail which blocks all merges.

**Acceptance criteria**

* It's possible to create a CentOS machine on Azure or we deprecate the CentOS support for Azure
* The E2E tests are not failing anymore

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Choose a different CentOS sku in azure, due to the removal of the current one.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #1040

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
Use CentOS sku 7_9 instead of 7-CI  
```

### CODE REVIEW

**Review by kron4eg — APPROVED** [review]

File: pkg/cloudprovider/provider/azure/provider.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ var imageReferences = map[providerconfigtypes.OperatingSystem]compute.ImageRefer
 	providerconfigtypes.OperatingSystemCentOS: {
 		Publisher: to.StringPtr("OpenLogic"),
 		Offer:     to.StringPtr("CentOS"),
-		Sku:       to.StringPtr("7-CI"), // https://docs.microsoft.com/en-us/azure/virtual-machines/linux/using-cloud-init
+		Sku:       to.StringPtr("7_9"), // https://docs.microsoft.com/en-us/azure/virtual-machines/linux/using-cloud-init
```
Comment (xmudrii) [inline pkg/cloudprovider/provider/azure/provider.go:124]: Should we re-enable CentOS tests on Azure now that CentOS should work?

https://github.com/kubermatic/machine-controller/blob/f1979a1d104fb88d3286307de576ec33c30a5c09/test/e2e/provisioning/all_e2e_test.go#L564-L565

**Review by xmudrii — APPROVED** [review]

**Review by xmudrii — APPROVED** [review]

**Pull request merged**